File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    
    int size = progresses.size();
    
    // q에 인덱스 번호를 기입
    for(int i = 0; i < size; i++) {
        q.push(i);
    }
    
    // 조건을 만족하면, 제거해주기
    while(!q.empty()) {
        int cnt = 0;
        // 100이 될 때까지 계속 더하기
        for (int i = 0; i < size; i++) {
            progresses[i] += speeds[i];
        }
        
        // q가 존재하고, 100 이상이 된다면, 제거 및 cnt 추가
        while(!q.empty() && progresses[q.front()] >= 100) {
            q.pop();
            cnt++;
        }
        
        if(cnt != 0) {
            answer.push_back(cnt);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    i = 0
    day = 0
    cnt = 0
    while(i < len(progresses)):
        if (progresses[i] + day * speeds[i] >= 100):
            cnt += 1
            i += 1
            if i == len(progresses):
                answer.append(cnt)
        else:
            if (cnt > 0):
                answer.append(cnt)
                cnt = 0
            day += 1
    return answer